### ISSUE TITLE
Compilation error

### ISSUE THREAD

**Issue by APB9785** [mathieuprog/tz_extra#3]
Adding Tz and Tz_extra (following the Tz readme) gives a compilation error when starting the application:
```
== Compilation error in file lib/compiler_runner.ex ==
** (ArgumentError) errors were found at the given arguments:

  * 1st argument: not a nonempty list

    :erlang.hd([])
    lib/compiler.ex:143: anonymous fn/1 in TzExtra.Compiler.add_offset_data/1
    (elixir 1.12.3) lib/enum.ex:1582: Enum."-map/2-lists^map/1-0-"/2
    (elixir 1.12.3) lib/enum.ex:1582: Enum."-map/2-lists^map/1-0-"/2
    lib/compiler.ex:20: TzExtra.Compiler.compile/0
    lib/compiler_runner.ex:6: (module)
could not compile dependency :tz_extra, "mix compile" failed. You can recompile this dependency with "mix deps.compile tz_extra", update it with "mix deps.update tz_extra" or clean it with "mix deps.clean tz_extra"
```

**Comment by mathieuprog** [comment]
It looks related to this: https://github.com/mathieuprog/tz/issues/8

**Comment by mathieuprog** [comment]
When I create a new project, add `tz_extra`, and compile, it works fine for me.

**Comment by mathieuprog** [comment]
Could you share your environment please?

**Comment by APB9785** [comment]
Sure
OS  10.14.6 (Mojave)
Elixir 1.12.3
Erlang 24.0.6

**Comment by mathieuprog** [comment]
It seems a problem concerning Mac machines it seems. But how could that be 😅

**Comment by APB9785** [comment]
```
defp add_offset_data(time_zones) do
    Enum.map(time_zones, fn %{time_zone: time_zone_id} = time_zone ->
      {:ok, periods} = Tz.PeriodsProvider.periods(time_zone_id)

      {utc_offset, dst_offset, zone_abbr, dst_zone_abbr} =
        case hd(periods) do
```
I'm wondering why would one of the time_zones return an empty list of periods?  Is the time_zone_id incorrect/missing?

I would look into the PeriodsProvider module but I can't seem to find it in the Tz repo.

**Comment by mathieuprog** [comment]
In Tz you have to look into compiler.ex, the module is being compiled in that file.

**Comment by mathieuprog** [comment]
Also it would be nice to add a failing test (as it works for me I can't make it fail).

**Comment by APB9785** [comment]
I just unit tested this pipeline where the error occurred
```
IanaFileParser.time_zones_with_country(countries)
|> add_time_zone_links(get_time_zone_links_for_canonical_fun)
|> add_offset_data()
|> Enum.sort_by(
  &{&1.country && normalize_string(&1.country.name), &1.utc_offset, &1.time_zone}
)
```
and everything passed OK.  Next I'll try it in a demo project and see if the error is still there.

**Comment by APB9785** [comment]
Error is still there.  tz + tz_extra in a new 1.6 project does not compile (same stacktrace as before).  Here is the repo: https://github.com/APB9785/tz_error_demo

**Comment by mathieuprog** [comment]
I think I might encounter the error now, uhmmm.. weird. I'll look into it tomorrow.

**Comment by mathieuprog** [comment]
The error is cause by the option `:reject_time_zone_periods_before_year`. Removing this option will prevent the error.
That will be a quick solution before I fix it.
Note that I don't even know if this option is worth it with the newer version of Tz because I think it might make no difference memory-wise. It needs to be measured (I just don't know how yet). Also this option was more intended for embedded devices that have a very limited amount of memory.

**Comment by mathieuprog** [comment]
Fixed in 0.20.1

**Comment by mathieuprog** [comment]
By the way `tz_extra` has a dependency on `tz` so you don't need `tz` in your `mix.exs` file

**Comment by APB9785** [comment]
The issue is indeed fixed for me by upgrading to 0.20.1

**Comment by OldhamMade** [comment]
Confirmed: fixed for me also. Thanks @mathieuprog.

FYI, here are some results from a *naïve* benchmark for my use-case on my local machine:

```text
Operating System: macOS
CPU Information: Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
Number of Available Cores: 16
Available memory: 64 GB
Elixir 1.12.0
Erlang 24.0.1

Benchmark suite executing with the following configuration:
warmup: 2 s
time: 30 s
memory time: 30 s
parallel: 4
inputs: none specified
Estimated total run time: 2.07 min

Benchmarking Tz...
Benchmarking Tzdata...

Name             ips        average  deviation         median         99th %
Tz            8.82 K      113.38 μs    ±52.02%       77.99 μs      324.99 μs
Tzdata        4.88 K      204.99 μs    ±17.63%      196.99 μs      352.99 μs

Comparison:
Tz            8.82 K
Tzdata        4.88 K - 1.81x slower +91.61 μs

Memory usage statistics:

Name           average  deviation         median         99th %
Tz            13.91 KB     ±0.09%       13.90 KB       13.95 KB
Tzdata        99.90 KB     ±0.09%       99.86 KB      100.09 KB

Comparison:
Tz            13.90 KB
Tzdata        99.90 KB - 7.18x memory usage +86.00 KB
```

My config is:

```elixir
config :elixir, :time_zone_database, Tz.TimeZoneDatabase
config :tz, reject_time_zone_periods_before_year: 2019
```
